sigstore-java/src/main/java/dev/sigstore/rekor/client/RekorVerifier.java

```diff
@@ -134,6 +138,43 @@ private void verifyInclusionProof(RekorEntry entry) throws RekorVerificationExce
     }
   }
 
+  private void verifyCheckpoint(RekorEntry entry, TransparencyLog tlog)
+      throws RekorVerificationException {
+    Checkpoint checkpoint;
+    try {
+      checkpoint = entry.getVerification().getInclusionProof().parsedCheckpoint();
+    } catch (RekorParseException ex) {
+      throw new RekorVerificationException("Could not parse checkpoint", ex);
+    }
+
+    byte[] inclusionRootHash =
+        Hex.decode(entry.getVerification().getInclusionProof().getRootHash());
+    byte[] checkpointRootHash = Base64.getDecoder().decode(checkpoint.getBase64Hash());
+
+    if (!Arrays.equals(inclusionRootHash, checkpointRootHash)) {
+      throw new RekorVerificationException(
+          "Checkpoint root hash does not match root hash provided in inclusion proof");
+    }
+    var keyHash = Hashing.sha256().hashBytes(tlog.getPublicKey().getRawBytes()).asBytes();
+    // checkpoint 0 is always the log, not any of the cross signing verifiers/monitors
+    var sig = checkpoint.getSignatures().get(0);
+    for (int i = 0; i < 4; i++) {
```

Comment:
what is the `4` for?

Reply:
Key ID is 4 bytes, https://github.com/sigstore/rekor/blob/main/pkg/util/signed_note.go#L140, truncated public key hash

Reply:
Would be good to have a link to that or https://github.com/C2SP/C2SP/blob/main/signed-note.md#signatures

Reply:
that should be added in a comment at least IMO

Reply:
It's defined in the spec and the spec is linked in the checkpoint type

Reply:
https://github.com/sigstore/sigstore-java/blob/main/sigstore-java/src/main/java/dev/sigstore/rekor/client/RekorEntry.java#L106